hepcrawl/parsers/arxiv.py

```diff
@@ -0,0 +1,363 @@
+            {'doi': value, 'material': 'publication'} for value in doi_values
+        ]
+
+        return dois
+
+    @property
+    def licenses(self):
+        licenses = self.root.xpath('.//license/text()').extract()
+        return [{'url':license, 'material': self.material} for license in licenses]
+
+    @property
+    def material(self):
+        return 'preprint'
+
+    @property
+    def number_of_pages(self):
+        comments = '; '.join(self.root.xpath('.//comments/text()').extract())
+        if not comments:
+            return None
+
+        found_pages = RE_PAGES.search(comments)
+        if found_pages:
+            pages = found_pages.group(1)
+            return maybe_int(pages)
+        
+        return None
+
+    @property
+    def publication_info(self):
+        publication_info = {
+            'material': 'publication',
+            'pubinfo_freetext': self.pubinfo_freetext,
+        }
+
+        return publication_info
+
+    @property
+    def pubinfo_freetext(self):
+        return self.root.xpath('.//journal-ref/text()').extract_first()
+
+    @property
+    def title(self):
+        return self.fix_long_text(self.root.xpath('.//title/text()').extract_first())
+
+    @staticmethod
+    def fix_long_text(text):
+        return re.sub('\s+', ' ', text).strip()
+
+    @staticmethod
+    def get_root_node(arxiv_record):
+        """Get a selector on the root ``article`` node of the record.
+
+        This can be overridden in case some preprocessing needs to be done on
+        the XML.
+
+        Args:
+            arxiv_record(Union[str, scrapy.selector.Selector]): the record in arXiv format.
+
+        Returns:
+            scrapy.selector.Selector: a selector on the root ``<article>``
+                node.
+        """
+        if isinstance(arxiv_record, six.string_types):
+            root = get_node(arxiv_record)
+        else:
+            root = arxiv_record
+        root.remove_namespaces()
+
+        return root
+
+    @property
+    def public_note(self):
+        comments = '; '.join(self.root.xpath('.//comments/text()').extract())
+
+        return comments
+
+    @property
+    def private_note(self):
+        _, _, warning = self.authors_and_collaborations
+
+        return warning
+
+    @property
+    def report_numbers(self):
+        report_numbers = self.root.xpath('.//report-no/text()').extract()
+        rns = []
+        for rn in report_numbers:
+            rns.extend(rn.split(', '))
+
+        return rns
+    
+    @property
+    def arxiv_eprint(self):
+        return self.root.xpath('.//id/text()').extract_first()
+
+    @property
+    def arxiv_categories(self):
+        categories = self.root.xpath('.//categories/text()').extract_first()
+
+        if categories:
+            return categories.split()
+
+        return []
+
+    @property
+    def document_type(self):
+        comments = '; '.join(self.root.xpath('.//comments/text()').extract())
+        if not comments:
```

Comment:
is this really needed? `comments` is always a string because of the `join`, so in the worst case the regex searches find nothing and you return `'article'` at the end.